File: solutions/free-pascal/line-up/1/LineUp.pas
unit LineUp;
{$mode ObjFPC}{$H+}
interface
function FormatMessage(
  const name : ShortString; const number : byte
) : ShortString;
implementation
uses SysUtils;
function FormatMessage(
  const name : ShortString; const number : byte
) : ShortString;
  function suffix : ShortString;
  begin
    if number div 10 mod 10 = 1 then
      result := 'th'
    else
      case number mod 10 of
        1: result := 'st';
        2: result := 'nd';
        3: result := 'rd';
      else
        result := 'th';
      end;
  end;
begin
  result := format('%s, you are the %d%s customer we serve today. Thank you!',
                   [name, number, suffix]);
end;
end.

File: solutions/python/line-up/2/line_up.py
"""Generate a customer queue message with the correct ordinal suffix."""


def line_up(name, number):
    """Return a formatted message containing the customer's ordinal position."""
    if 11 <= number % 100 <= 13:
        suffix = "th"
    else:
        suffix = {
            1: "st",
            2: "nd",
            3: "rd",
        }.get(number % 10, "th")

    return (
        f"{name}, you are the {number}{suffix} customer "
        "we serve today. Thank you!"
    )